vm/devices/storage/disk_backend/src/lib.rs

```diff
@@ -117,210 +135,256 @@ pub trait SimpleDisk: Send + Sync + Inspect + AsyncDisk {
     }
 
     /// Issues an asynchronous eject media operation to the disk.
-    fn eject(&self) -> StackFuture<'_, Result<(), DiskError>, { ASYNC_DISK_STACK_SIZE }> {
-        StackFuture::from(ready(Err(DiskError::UnsupportedEject)))
-    }
-}
-
-impl SimpleDisk for Arc<dyn SimpleDisk> {
-    fn disk_type(&self) -> &str {
-        self.as_ref().disk_type()
+    fn eject(&self) -> impl Future<Output = Result<(), DiskError>> + Send {
+        ready(Err(DiskError::UnsupportedEject))
     }
 
-    fn sector_count(&self) -> u64 {
-        self.as_ref().sector_count()
-    }
-
-    fn sector_size(&self) -> u32 {
-        self.as_ref().sector_size()
-    }
+    /// Issues an asynchronous read-scatter operation to the disk.
+    ///
+    /// # Arguments
+    /// * `buffers` - An object representing the data buffers into which the disk data will be transferred.
+    /// * `sector` - The logical sector at which the read operation starts.
+    fn read_vectored(
+        &self,
+        buffers: &RequestBuffers<'_>,
+        sector: u64,
+    ) -> impl Future<Output = Result<(), DiskError>> + Send;
 
-    fn disk_id(&self) -> Option<[u8; 16]> {
-        self.as_ref().disk_id()
-    }
+    /// Issues an asynchronous write-gather operation to the disk.
+    /// # Arguments
+    /// * `buffers` - An object representing the data buffers containing the data to transfer to the disk.
+    /// * `sector` - The logical sector at which the write operation starts.
+    /// * `fua` - A flag indicates if FUA (force unit access) is requested.
+    fn write_vectored(
+        &self,
+        buffers: &RequestBuffers<'_>,
+        sector: u64,
+        fua: bool,
+    ) -> impl Future<Output = Result<(), DiskError>> + Send;
 
-    fn physical_sector_size(&self) -> u32 {
-        self.as_ref().physical_sector_size()
-    }
+    /// Issues an asynchronous flush operation to the disk.
+    fn sync_cache(&self) -> impl Future<Output = Result<(), DiskError>> + Send;
 
-    fn is_fua_respected(&self) -> bool {
-        self.as_ref().is_fua_respected()
+    /// Waits for the disk sector size to be different than the specified value.
+    fn wait_resize(&self, sector_count: u64) -> impl Future<Output = u64> + Send {
+        let _ = sector_count;
+        std::future::pending()
     }
+}
 
-    fn is_read_only(&self) -> bool {
-        self.as_ref().is_read_only()
+/// An asynchronous block device.
+///
+/// This type is cheap to clone, for sharing the disk among multiple concurrent
+/// users.
+#[derive(Inspect, Clone)]
+#[inspect(extra = "Self::inspect_extra")]
+pub struct SimpleDisk(#[inspect(flatten)] Arc<SimpleDiskInner>);
```

Comment:
while I'm sure we've all grown to love and appreciate the name SimpleDisk (so simple! much easy! wow!), maybe this is the PR where we switch it over to a more representative name?

after all - it seems unlikely that we'll find ourselves in a situation where we have another distinct frontend around `DiskIo`, right? i.e: there's no `AdvancedDisk` coming on the horizon, is there?

Reply:
Alright. Let's just call it `Disk`.